Rules/CommonScripts/GameStateBanners.as

```diff
@@ -0,0 +1,189 @@
+#define CLIENT_ONLY
+
+Vec2f bannerStart = Vec2f_zero;
+Vec2f bannerPos = Vec2f_zero;
+Vec2f bannerDest = Vec2f_zero;
+f32 frameTime = 0;
+const f32 maxTime = 0.6f;
+
+bool minimap = true;
+
+const u32 winBannerDuration = 8 * getTicksASecond();
+const u32 buildBannerDuration = 5 * getTicksASecond();
+const u32 gameBannerDuration = 5 * getTicksASecond();
+
+void onInit(CRules@ this)
+{
+	if (!GUI::isFontLoaded("AveriaSerif-Bold_32"))
+	{
+		string AveriaSerif = CFileMatcher("AveriaSerif-Bold.ttf").getFirst();
+		GUI::LoadFont("AveriaSerif-Bold_32", AveriaSerif, 32, true);
+	}
+
+	minimap = this.minimap;
+	onRestart(this);
+}
+
+void onRestart(CRules@ this)
+{
+	ResetBannerInfo(this);
+	this.minimap = minimap;
+
+	SetBanner(this);
+}
+
+void onNewPlayerJoin(CRules@ this, CPlayer@ player)
+{
+	if (getLocalPlayer() !is null && getLocalPlayer().getUsername() == player.getUsername())
+	{
+		SetBanner(this);
+	}
+}
+
+void onStateChange(CRules@ this, const u8 oldState)
+{
+	ResetBannerInfo(this);
+	SetBanner(this);
+}
+
+void onTick(CRules@ this)
+{
+	if (this.get_string("Animate Banner") != "none" && this.get_u32("Banner End") < getGameTime())
+	{
+		ResetBannerInfo(this);
+	}
+}
+
+void onRender(CRules@ this)
+{
+	if (this.get_string("Animate Banner") != "none")
```

Comment:
Instead of getting the property here, as well as on line 71, 77 and 81, just store it once before this if-statement/

Reply:
fixed